pkg/reduce/data_forward.go

```diff
@@ -179,21 +179,21 @@ func (df *DataForward) ReplayPersistedMessages(ctx context.Context) error {
 // forwardAChunk reads a chunk of messages from isb and assigns watermark to messages
 // and writes the messages to pbq
 func (df *DataForward) forwardAChunk(ctx context.Context) {
-	readMessages, err := df.fromBuffer.Read(ctx, df.opts.readBatchSize)
+	readMessages, err := df.fromBufferPartition.Read(ctx, df.opts.readBatchSize)
 	totalBytes := 0
 	if err != nil {
 		df.log.Errorw("Failed to read from isb", zap.Error(err))
 		readMessagesError.With(map[string]string{
 			metrics.LabelVertex:             df.vertexName,
 			metrics.LabelPipeline:           df.pipelineName,
 			metrics.LabelVertexReplicaIndex: strconv.Itoa(int(df.vertexReplica)),
-			"buffer":                        df.fromBuffer.GetName()}).Inc()
+			"partition_idx":                 df.fromBufferPartition.GetName()}).Inc()
```

Comment:
please review the latest changes.

Reply:
Based on the latest change, it seems the answer to my question is YES. Could you confirm?

Reply:
vertex replica and partition_idx will have one-to-one mapping while reading while writing its one to many.

Reply:
By `while writing`, do you mean that one replica writing to multiple out-going partitions? I thought in this metric, the partition_idx means in-coming partitions.

Reply:
yes, one replica writing to multiple toBufferPartitions.  I updated `parition_idx` to `partition_name`.

Reply:
Since this metric is `readMessagesError`, `LabelVertexReplicaIndex ` and `partition_name` are 1:1. It seems unnecessary to adding the extra dimension for reduce.

Reply:
I prefer keeping both because we will be able to filter either using the replica or the stream name.